IllegalArgumentException when instantiating a ProxyConfiguration in DataBoundConfigurator.describe on line 219

Following up on #229, we've got another error when triggering an export, when some proxy info exist on `/jenkins/pluginManager/advanced`.

Thanks @victorsalaun for spotting this :)

This is the stacktrace we got :
```
[WARNING] Error while serving http://localhost:8080/jenkins/configuration-as-code/export
java.lang.reflect.InvocationTargetException
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:347)
	at org.kohsuke.stapler.interceptor.RequirePOST$Processor.invoke(RequirePOST.java:52)
	at org.kohsuke.stapler.PreInvokeInterceptedFunction.invoke(PreInvokeInterceptedFunction.java:26)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:184)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:117)
	at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:129)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
	at org.kohsuke.stapler.MetaClass$10.dispatch(MetaClass.java:374)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:135)
	at jenkins.metrics.impl.MetricsFilter.doFilter(MetricsFilter.java:125)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:132)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:138)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:564)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
	at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.jenkinsci.plugins.casc.DataBoundConfigurator.describe(DataBoundConfigurator.java:219)
	at org.jenkinsci.plugins.casc.plugins.PluginManagerConfigurator.describe(PluginManagerConfigurator.java:257)
	at org.jenkinsci.plugins.casc.plugins.PluginManagerConfigurator.describe(PluginManagerConfigurator.java:49)
	at org.jenkinsci.plugins.casc.ConfigurationAsCode.doExport(ConfigurationAsCode.java:232)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:625)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
	... 63 more
```

--- a/src/main/java/org/jenkinsci/plugins/casc/Attribute.java
+++ b/src/main/java/org/jenkinsci/plugins/casc/Attribute.java
@@ -1,11 +1,13 @@
 package org.jenkinsci.plugins.casc;
 
 import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang.reflect.FieldUtils;
 import org.jenkinsci.plugins.casc.model.CNode;
 import org.jenkinsci.plugins.casc.model.Sequence;
 
 import java.beans.PropertyDescriptor;
 import java.lang.reflect.Array;
+import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -167,7 +169,12 @@ public class Attribute<Owner, Type> {
 
     private Type _getValue(Owner target) throws Exception {
         final PropertyDescriptor property = PropertyUtils.getPropertyDescriptor(target, name);
-        if (property == null) return null;
+        if (property == null) {
+            // If this is a public final field, developers don't define getters as jelly can use them as-is
+            final Field field = FieldUtils.getField(target.getClass(), name);
+            if (field != null) return (Type) field.get(target);
+            return null;
+        }
         final Method readMethod = property.getReadMethod();
         if (readMethod == null) return null;
         return (Type) readMethod.invoke(target);

--- a/src/main/java/org/jenkinsci/plugins/casc/Attribute.java
+++ b/src/main/java/org/jenkinsci/plugins/casc/Attribute.java
@@ -8,6 +8,7 @@ import org.jenkinsci.plugins.casc.model.Sequence;
 import java.beans.PropertyDescriptor;
 import java.lang.reflect.Array;
 import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -167,17 +168,21 @@ public class Attribute<Owner, Type> {
         T getValue(O target) throws Exception;
     }
 
-    private Type _getValue(Owner target) throws Exception {
-        final PropertyDescriptor property = PropertyUtils.getPropertyDescriptor(target, name);
-        if (property == null) {
+    private Type _getValue(Owner target) throws ConfiguratorException {
+        try {
+            final PropertyDescriptor property = PropertyUtils.getPropertyDescriptor(target, name);
+            if (property != null && property.getReadMethod() != null) {
+                return (Type) property.getReadMethod().invoke(target);
+            }
             // If this is a public final field, developers don't define getters as jelly can use them as-is
             final Field field = FieldUtils.getField(target.getClass(), name);
-            if (field != null) return (Type) field.get(target);
-            return null;
+            if (field == null) {
+                throw new ConfiguratorException("Can't read attribute '" + name + "' from "+ target);
+            }
+            return (Type) field.get(target);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            throw new ConfiguratorException("Can't read attribute '" + name + "' from "+ target, e);
         }
-        final Method readMethod = property.getReadMethod();
-        if (readMethod == null) return null;
-        return (Type) readMethod.invoke(target);
     }
 
     /**

--- a/src/main/java/org/jenkinsci/plugins/casc/ObsoleteConfigurationMonitor.java
+++ b/src/main/java/org/jenkinsci/plugins/casc/ObsoleteConfigurationMonitor.java
@@ -51,7 +51,8 @@ public class ObsoleteConfigurationMonitor extends AdministrativeMonitor {
     }
 
     public String getCss() {
-        if (Jenkins.getVersion().isNewerThan(new VersionNumber("2.103"))) {
+        final VersionNumber version = Jenkins.getVersion();
+        if (version == null || version.isNewerThan(new VersionNumber("2.103"))) {
             return "alert alert-warning";
         }
         return "warning";

--- a/src/main/java/org/jenkinsci/plugins/casc/model/Source.java
+++ b/src/main/java/org/jenkinsci/plugins/casc/model/Source.java
@@ -1,9 +1,13 @@
 package org.jenkinsci.plugins.casc.model;
 
+import java.io.Serializable;
+
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
-public class Source {
+public class Source implements Serializable {
+
+    static final long serialVersionUID = 1L;
 
     public final String file;
     public final int line;

